### PR TITLE
feat(utils): add robust SplitList utility with unit tests (Fixes #555)

### LINKED ISSUE #555: Logic for splitting comma-separated inputs need to be fixed
**What happened**: The comma-separated inputs are usually validated using a similar paradigm across all experiments:

```go
instanceNamesList := strings.Split(experimentsDetails.VMInstanceName, ",")
if len(instanceNamesList) == 0 {
	return errors.Errorf("no instance name found to stop")
}
```

However, if the input string is an empty string then still the length of the resultant slice will be 1. Hence this check maybe simply updated with the following in all places:

```go
if experimentsDetails.VMInstanceName == "" {
        return errors.Errorf("no instance name found to stop")
}
```

### PR DESCRIPTION
**Pull request by xnaveenx**
What this PR does / why we need it:

This PR introduces a centralized helper function SplitList in pkg/utils/stringutils to safely handle comma-separated strings. It addresses the logic bug described in #555 where empty strings incorrectly resulted in a slice of length 1.

Improvements over previous attempts: This PR supersedes stale PRs #564 and #686 by:

Centralizing Logic: Instead of fixing individual experiment files inline, this provides a reusable, tested function.

Edge Case Handling: Automatically trims spaces ("a, b" -> ["a","b"]) and filters empty elements ("a,,b" -> ["a","b"]).

Robust Testing: Includes unit tests covering edge cases like inputs containing only commas (,,,), satisfying the feedback left on PR #686.

Which issue this PR fixes:

fixes #555

Special notes for your reviewer:

Strategy: I have purposefully limited this PR to the utility implementation and unit tests to keep the review atomic and clean. Once this is merged, I will open a follow-up "refactor" PR to update the experiment files (e.g., pod_delete.go) to use this new SplitList function.

Checklist:

[x] Fixes #555

[ ] PR messages has document related information

[ ] Labelled this PR & related issue with breaking-changes tag

[ ] PR messages has breaking changes related information

[ ] Labelled this PR & related issue with requires-upgrade tag

[ ] PR messages has upgrade related information

[x] Commit has unit tests

[ ] Commit has integration tests

[ ] E2E run Required for the changes

### CODE REVIEW

File: pkg/utils/stringutils/string.go [hunk 1]
```diff
@@ -12,6 +13,21 @@ const (
 	letterIdxMax  = 63 / letterIdxBits   // # of letter indices fitting in 63 bits
 )
 
+func SplitList(s string) []string {
+	if s == "" {
```
Comment (ispeakc0de) [inline pkg/utils/stringutils/string.go]:
```suggestion
	if strings.TrimSpace(s) == "" {
```

File: pkg/utils/stringutils/string.go [hunk 2]
```diff
@@ -12,6 +13,21 @@ const (
...
+		return []string{}
+	}
+	parts := strings.Split(s, ",")
```
Comment (ispeakc0de) [inline pkg/utils/stringutils/string.go]:
```suggestion
	parts := strings.Split(strings.TrimSpace(s), ",")
```

**Review by ispeakc0de — APPROVED** [review]
